### PR TITLE
VPN Proxy stop and pixel improvements

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/0/0/1208270696831728/f

## Description

- Fixes a bug that caused the proxy to not be relaunched when stopped.
- Several improvements to pixels to unify naming and introduce an additional pixel.
- Several improvements to modularize the logging and make it more privacy-aware.

### Testing: Proxy relaunches after manually stopping it

1. Open System Settings > Control Center > VPN
2. Enable "Show in menu bar", see that there's a System VPN menu in your status bar
3. Go back to our browser and [enable PrivacyPro](https://app.asana.com/0/1207603085593419/1207158980612262/f)
4. Start the VPN
5. Log in as an internal user through use-login.duckduckgo.com
6. Go to any site
7. Click on the in-browser VPN icon to show the status view
8. Exclude the site from the VPN
9. Using the system VPN status bar menu, check that the VPN proxy is launched
10. Stop it manually and wait ~1 second
11. Check again and see that it's running again
12. Remove the site exclusion from the VPN
13. Check that the proxy is stopped

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: DuckDuckGo/NetworkProtection/AppTargets/BothAppTargets/NetworkProtectionTunnelController.swift [hunk 1]
```diff
@@ -84,7 +84,7 @@ final class NetworkProtectionTunnelController: TunnelController, TunnelSessionPr
 
     private let notificationCenter: NotificationCenter
 
-    /// The proxy manager
+    /// The tunnel manager
```
Comment (diegoreymendez) [inline DuckDuckGo/NetworkProtection/AppTargets/BothAppTargets/NetworkProtectionTunnelController.swift:87]: Just found a typo.  The documentation here is for the tunnel not the proxy.

File: DuckDuckGo/NetworkProtection/NetworkExtensionTargets/NetworkExtensionTargets/MacTransparentProxyProvider.swift [hunk 2]
```diff
@@ -85,9 +79,13 @@ final class MacTransparentProxyProvider: TransparentProxyProvider {
             }
         }
 #endif
+
+        super.init(settings: settings,
+                   configuration: configuration,
+                   logger: Self.vpnProxyLogger)
```
Comment (diegoreymendez) [inline DuckDuckGo/NetworkProtection/NetworkExtensionTargets/NetworkExtensionTargets/MacTransparentProxyProvider.swift:85]: Let's initialize the proxy after PixelKit.

File: DuckDuckGoVPN/VPNProxyLauncher.swift [hunk 3]
```diff
@@ -89,12 +89,16 @@ final class VPNProxyLauncher {
 
             isControllingProxy = true
 
+            defer {
+                isControllingProxy = false
+            }
```
Comment (diegoreymendez) [inline DuckDuckGoVPN/VPNProxyLauncher.swift:94]: It's safer to defer this here than to try and make this false in every exit.

File: DuckDuckGo.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1546,6 +1546,13 @@
 		566B736A2BECC02D00FF1959 /* SyncAlertsPresenter.swift in Sources */ = {isa = PBXBuildFile; fileRef = 566B73682BECBF8400FF1959 /* SyncAlertsPresenter.swift */; };
 		566B736C2BECC3C600FF1959 /* SyncPausedStateManaging.swift in Sources */ = {isa = PBXBuildFile; fileRef = 566B736B2BECC3C600FF1959 /* SyncPausedStateManaging.swift */; };
 		566B736D2BECC3C600FF1959 /* SyncPausedStateManaging.swift in Sources */ = {isa = PBXBuildFile; fileRef = 566B736B2BECC3C600FF1959 /* SyncPausedStateManaging.swift */; };
+		567A23BE2C7F539C0010F66C /* SpecialErrorPageUserScriptExtension.swift in Sources */ = {isa = PBXBuildFile; fileRef = 567A23BD2C7F539C0010F66C /* SpecialErrorPageUserScriptExtension.swift */; };
```
Comment (diegoreymendez) [inline DuckDuckGo.xcodeproj/project.pbxproj:1549]: This is just Xcode being Xcode.  These files are moved around within the pbxproj file.

File: LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionProxy/Provider/TransparentProxyProvider.swift [hunk 5]
```diff
@@ -153,61 +197,49 @@ open class TransparentProxyProvider: NETransparentProxyProvider {
...
+        eventHandler.handle(event: .startAttempt(.begin))
 
         do {
-            logger.log(
-                """
-                Starting proxy\n
-                > configuration: \(String(describing: self.configuration), privacy: .public)\n
-                > settings: \(String(describing: self.settings), privacy: .public)\n
-                > options: \(String(describing: options), privacy: .public)
-                """)
```
Comment (diegoreymendez) [inline LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionProxy/Provider/TransparentProxyProvider.swift:165]: Removed due to not being privacy-aware.

**Review by samsymons — APPROVED** [review]
Testing steps worked for me, and the changes all look sensible - thanks for the logging improvements etc. in here as well.

**Comment by diegoreymendez** [comment]
Thanks @samsymons !

**Pull request merged**